### PR TITLE
Make sure PyQt5 is supported

### PR DESCRIPTION
**Pull request by astrofrog**
To install in conda: http://stackoverflow.com/questions/25468397/using-anaconda-python-with-pyqt5

### CODE REVIEW

**Comment by astrofrog** [comment]
Stuck on this very frustrating error:

```python
glue/qt/widgets/mpl_widget.py:38: in wrapper
    result = func(*args, **kwargs)
glue/core/application_base.py:72: in new_data_viewer
    c = viewer_class(self._session)
glue/qt/widgets/histogram_widget.py:46: in __init__
    super(HistogramWidget, self).__init__(session, parent)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <glue.qt.widgets.histogram_widget.HistogramWidget object at 0x12f035180>
session = <glue.core.session.Session object at 0x12e99a4d0>, parent = None

    def __init__(self, session, parent=None):
        """
            :type session: :class:`~glue.core.Session`
            """
>       QMainWindow.__init__(self, parent)
E       TypeError: __init__() takes exactly 2 arguments (1 given)
```

Even though I *am* passing two arguments. However it's possible parent no longer takes ``None`` as an argument but that still wouldn't explain error message

**Comment by astrofrog** [comment]
So inverting the order of inheritance for ``DataViewer`` to:

```python
class DataViewer(ViewerBase, QMainWindow):
```

fixed that issue, somehow. Not 100% I understand at this point.

**Comment by astrofrog** [comment]
Only two failing tests...

**Comment by astrofrog** [comment]
Need to set the following to get HiDPI icons:

```python
app.setAttribute(QtCore.Qt.AA_UseHighDpiPixmaps);
```

**Comment by astrofrog** [comment]
Main to-dos:

* [ ] Move changes from ``glue.external.qt`` to ``qt_helpers``
* [ ] Check how icons look on non-HiDPI display
* [ ] Get all tests passing
* [ ] Check that we don't break anything on PyQt4

**Comment by astrofrog** [comment]
The following will be required for Matplotlib icons to look good in PyQt5: https://github.com/matplotlib/matplotlib/pull/5057

File: glue/qt/data_collection_model.py [hunk 1]
```diff
@@ -243,7 +244,11 @@ def __init__(self, data_collection, parent=None):
         # without this reference, PySide clobbers instance
         # data of model items
         self.register_to_hub(self.data_collection.hub)
-        self.setSupportedDragActions(Qt.CopyAction)
+        if not is_pyqt5():
+            self.setSupportedDragActions(Qt.CopyAction)
```
Comment (ChrisBeaumont) [inline glue/qt/data_collection_model.py]: Does Qt5 not support this?

File: glue/qt/ui/layertree.py [hunk 2]
```diff
@@ -74,8 +74,15 @@ def setupUi(self, LayerTree):
         QtCore.QMetaObject.connectSlotsByName(LayerTree)
 
     def retranslateUi(self, LayerTree):
-        LayerTree.setWindowTitle(QtGui.QApplication.translate("LayerTree", "Form", None, QtGui.QApplication.UnicodeUTF8))
-        self.layerAddButton.setToolTip(QtGui.QApplication.translate("LayerTree", "Load a new data set", None, QtGui.QApplication.UnicodeUTF8))
-        self.newSubsetButton.setToolTip(QtGui.QApplication.translate("LayerTree", "Create a new empty subset", None, QtGui.QApplication.UnicodeUTF8))
-        self.layerRemoveButton.setToolTip(QtGui.QApplication.translate("LayerTree", "Delete Layer", None, QtGui.QApplication.UnicodeUTF8))
-        self.linkButton.setToolTip(QtGui.QApplication.translate("LayerTree", "Link data", None, QtGui.QApplication.UnicodeUTF8))
+        if is_pyqt5():
```
Comment (ChrisBeaumont) [inline glue/qt/ui/layertree.py]: nit: this would be cleaner as a translate helper function

File: glue/qt/data_collection_model.py (hunk 1 above)
Comment (astrofrog) [inline glue/qt/data_collection_model.py]: It does but I had to add a method now, see a few lines down.

File: glue/qt/ui/layertree.py (hunk 2 above)
Comment (astrofrog) [inline glue/qt/ui/layertree.py]: @ChrisBeaumont - just to check, could you clarify what you mean here?

Comment (ChrisBeaumont) [inline glue/qt/ui/layertree.py]: nvm, I think I misread the diff. This is fine

**Comment by astrofrog** [comment]
There were issues with the existing PyQt5 packages for conda, so I made a Docker image in which to build my own: https://hub.docker.com/r/astrofrog/ubuntu-12.04-conda-pyqt5-builder/ - the conda package can be installed with ``conda install -c astrofrog pyqt5``. Fingers crossed that it works.

**Comment by astrofrog** [comment]
lotr-victory

**Comment by astrofrog** [comment]
Well, this took a while :-/

:tada:

**Comment by ChrisBeaumont** [comment]
Nicely done! 

image

**Pull request merged**